src/Routes/seasons.ts

```diff
@@ -0,0 +1,31 @@
+﻿/*
+ * Seasons controller
+ */
+//import * as assert from "assert";
+import * as express from "express";
+import { SqlDataProvider } from "../DataProviders/SqlDataProvider/SqlDataProvider";
+import { Configuration } from "../Common/Configuration";
+import { Networking } from "../Common/Helpers";
+import { IDataProvider } from "../DataProviders/IDataProvider";
+const router = express.Router();
+
+// TODO: Add a factory pattern
+const dataProvider: IDataProvider = new SqlDataProvider(Configuration.sqlConfig);
+
+router.get('/', (req: express.Request, res: express.Response) => {
+    dataProvider.getSeasonList()
```

Comment:
`async..await` will make your life easier here.

Reply:
In general though, should I be using awaits or then?  Or does that depend?